dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+
+        } finally {
+            DbConnectionFactory.closeSilently();
+        }
+
+    }
+
+    public void init(ServletConfig config) throws ServletException {
+
+
+
+    }
+
+    public void doPreviewMode(HttpServletRequest request, HttpServletResponse response) throws Exception {
+
+        String uri = URLDecoder.decode(request.getRequestURI(), UtilMethods.getCharsetConfiguration());
+        PageMode mode = PageMode.PREVIEW;
+
+        Host host = hostWebAPI.getCurrentHost(request);
+
+
+        // Getting the user to check the permissions
+        User user = com.liferay.portal.util.PortalUtil.getUser(request);
+
+        // Getting the identifier from the uri
+        Identifier id = APILocator.getIdentifierAPI()
+            .find(host, uri);
+        request.setAttribute("idInode", id.getId());
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY HTML INODE=" + id.getId());
+
+        Template template = null;
+        Template hostVariablesTemplate = null;
+
+        // creates the context where to place the variables
+        response.setContentType(CHARSET);
+        Context context = VelocityUtil.getWebContext(request, response);
+
+        IHTMLPage htmlPage = VelocityUtil.getPage(id, request, false, context);
+        if ("contentlet".equals(htmlPage.getType())) {
+            context.put("dotPageContent", new ContentMap(((Contentlet) htmlPage), user, true, host, context));
+        }
+
+
+
+        com.dotmarketing.portlets.templates.model.Template cmsTemplate = APILocator.getHTMLPageAssetAPI()
+            .getTemplate(htmlPage, true);
+
+
+
+        VelocityUtil.makeBackendContext(context, htmlPage, cmsTemplate.getInode(), id.getURI(), request, mode, host);
+
+
+
+        PageContextBuilder builder = new PageContextBuilder(htmlPage, user, PageMode.PREVIEW);
+        builder.addAll(context);
+
+
+
+        hostVariablesTemplate = VelocityUtil.getEngine()
+            .getTemplate(PageMode.PREVIEW.name() + File.separator + host.getIdentifier() + "." + VelocityType.SITE.fileExtension);
+
+        if (cmsTemplate.isDrawed()) {// We have a designed template
+            // Setting some theme variables
+            Map<String, Object> dotThemeData = DotTemplateTool.theme(cmsTemplate.getTheme(), host.getIdentifier());
+            context.put("dotTheme", dotThemeData);
+            context.put("dotThemeLayout", DotTemplateTool.themeLayout(cmsTemplate.getInode()));
+            // Our designed template
+            template = VelocityUtil.getEngine()
+                .getTemplate((String) dotThemeData.get("templatePath"));
+        } else {
+            template = VelocityUtil.getEngine()
+                .getTemplate(PageMode.PREVIEW.name() + File.separator + cmsTemplate.getIdentifier() + "."
+                        + VelocityType.TEMPLATE.fileExtension);
+        }
+
+
+
+        PrintWriter out = response.getWriter();
+        request.setAttribute("velocityContext", context);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'VELOCITY_CONTEXT' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)